bin/utest/signal.c

```diff
@@ -295,6 +296,93 @@ TEST_ADD(signal_sigsuspend) {
   return 0;
 }
 
+/* ======= signal_sigtimedwait ======= */
+int sigtimedwait_child(void *arg) {
+  pid_t ppid = getppid();
+  sched_yield();
+  syscall_ok(kill(ppid, SIGUSR1));
```

Comment:
I think that you can't assume that parent will do all initialization needed before sending him the `SIGUSR1` signal. Please look at other tests in that file and check if you can synchronize using other signals (I think you can use similar mechanism to that in `signal_stop` test.)